### PR TITLE
[jit] remove the addmm decomposation

### PR DESCRIPTION
**Pull request by wanchaol**
This PR removes addmm decomposation in the canonicalize_ops.cpp, this is mainly because of two reasons:

1. The original motivation for addmm decomposation is not valid anymore. Right now in our autodiff and shape_analysis we already include the `addmm` op
2. The `addmm` is never get fused again later in the pass. This is because we manually disabled it sometime ago. Also we really want `addmm` instead of `mm + add` for multiple backends integration.

### CODE REVIEW

**Comment by wanchaol** [comment]
turned out it triggered the batchmm optimization and make the LSTM regressed, will drop this PR and pursue another approach

**Comment by apaszke** [comment]
One way to tackle this would be to postpone the decomposition until the fuser, similarly to how we handle batch normalization.

**Comment by wanchaol** [comment]
@apaszke yeah that's a possible way. The problem with that is the `batchmm` pass happens before fuser, if we delay the decomposition to fuser, some `batchmm` optimization will not be triggered as well.. I am thinking if we should separate the decomposition into a separate pass before `batchmm`, maybe just merge it with canonicalize_ops.cpp. but this will not trigger the autodiff formulas for `batchnorm` and other ops that will be decomposed.

**Comment by apaszke** [comment]
Or maybe we can make mm batching recognize addmm as a valid pattern for add + mm too?

**Comment by wanchaol** [comment]
> Or maybe we can make mm batching recognize addmm as a valid pattern for add + mm too?

While mm batching can recognize the pattern for addmm, but it will erase the helpful high level node information. I am doing additional changes to make sure different backends can know the high level operator information before custom fusion. A viable way is: 1. lower mm batching to after custom fusion but before default fusion 2. split out the decomposition from canonicalize_ops and fuser to a separate pass after custom fusion but before batchmm.

**Comment by wanchaol** [comment]
@apaszke I implemented what I purposed above starting in #19987, let me know your thoughts :)
